tests/integration/embed/embed_test.go

```diff
@@ -210,3 +211,17 @@ func setupEmbedCfg(cfg *embed.Config, curls []url.URL, purls []url.URL) {
 	}
 	cfg.InitialCluster = cfg.InitialCluster[1:]
 }
+
+func TestEmbedEtcdAutoCompactionRetentionRetained(t *testing.T) {
+	cfg := embed.NewConfig()
+	urls := newEmbedURLs(false, 2)
+	setupEmbedCfg(cfg, []url.URL{urls[0]}, []url.URL{urls[1]})
+	cfg.Dir = filepath.Join(t.TempDir(), "embed-etcd")
+
+	cfg.AutoCompactionRetention = "2"
+
+	e, _ := embed.StartEtcd(cfg)
+	autoCompactionRetention := e.Server.Cfg.AutoCompactionRetention
+	duration_to_compare, _ := time.ParseDuration("2h0m0s")
+	assert.Equal(t, duration_to_compare, autoCompactionRetention)
```

Comment:
```suggestion
	assert.Equal(t, duration_to_compare, autoCompactionRetention)
	e.Close()
```

It's generally a best practice to close the server before the test ends. Otherwise, you will have some weird test failures like the following 
```
{
    "level": "fatal",
    "ts": "2023-05-02T18:55:06.775177Z",
    "caller": "etcdserver/server.go:614",
    "msg": "failed to purge wal file",
    "error": "open /tmp/TestEmbedEtcdAutoCompactionRetentionRetained1101367192/001/embed-etcd/member/wal: no such file or directory",
    "stacktrace": "go.etcd.io/etcd/server/v3/etcdserver.(*EtcdServer).purgeFile\n\t/home/etcd/server/etcdserver/server.go:614\ngo.etcd.io/etcd/server/v3/etcdserver.(*EtcdServer).GoAttach.func1\n\t/home/etcd/server/etcdserver/server.go:2425"
}
```

It has been proved in the CI test failures. Please consider fix this ~

Reply:
You can try comment line 203 which suppresses the log output `cfg.LogOutputs = []string{"/dev/null"}` locally.